### PR TITLE
fix error on picking latest vault releases

### PR DESCRIPTION
**Pull request by markmartirosian**
The version `1.2.6.1+ent` is not valid semver in https://releases.hashicorp.com/vault/index.json.

### CODE REVIEW

**Comment by dbanck** [comment]
Hi @markmartirosian! Thanks a lot for catching this bug and raising this PR!

I've created some test cases for `getRelease` in https://github.com/hashicorp/js-releases/pull/58 to catch and prevent bugs like this in the future. Do you mind rebasing the latest changes from `main` and adding a test case for versions like `1.2.6.1+ent`? Something like `it('should filter invalid versions', ...`. I'm happy to assist you with this or add the changes to your branch so that we can get this merged.

**Comment by markmartirosian** [comment]
@dbanck something like this? (;

File: src/index.test.ts [hunk 1]
```diff
@@ -111,4 +111,22 @@ describe('getRelease', () => {
...
+      name,
+      versions: {
+        '1.2.6.1+ent': { name, version: '1.2.6.1+ent' },
+        '1.2.7': { name, version: '1.2.7' },
+        '1.5.0': { name, version: '1.5.0' },
+      },
+    }));
+
+    const version = '1.2.7';
+    const release = await getRelease(name, version);
```
Comment (dbanck) [inline src/index.test.ts:126]: If we want to test the case you raised this PR for - selecting the latest version - we have to omit `version` here.

**Review by dbanck — COMMENTED** [review]
Thanks for updating the PR with the changes from `main` and providing a test case!

Can you explain what your motivation is behind extending your solution? I think the one-line filter was an elegant solution as a bug fix.

Do you mind removing the changes to `package-lock.json`? Unfortunately, this project still uses an older version of npm, so the lock file is still version 1. I'm planning to update this in a separate PR soon.

**Comment by markmartirosian** [comment]
@dbanck both codepaths: 1. `new Release` and 2: `matchVersion` depend on a `versions: Record<string, Release>`, this changes ensure that it's the same object in both cases. It's a little defensive, but ultimately reduces complexity and possibility of another bug given that the `matchVersion` uses `semver.maxSatisfying` on the `versions` object.

I've copied `package-lock.json` from the main branch. It's still showing changes for some reason.

You should have write permissions to the fork repo.

**Comment by dbanck** [comment]
Thanks for the write access! I rebased from `main` & force pushed to get rid of the changes in `package-lock.json`.

Unifying the versions for both code paths makes sense! I've added a new type `ResponseVersion` for the JSON response, because we can't use `Release` here. And one more test case when passing an invalid version.

**Review by dbanck — APPROVED** [review]
Thanks again for reporting and fixing this bug! 👍

**Pull request merged**